### PR TITLE
Update stm32f4 SDIO register fields and values

### PR DESCRIPTION
**Pull request by jkristell**
Documentation and limits for the sdio register values.

### CODE REVIEW

**Comment by jkristell** [comment]
I only added it to the f405 for now. But if this change looks OK to you, I can add the sdio peripheral to all other f4 devices as well. Just wanted to make sure it was the right approach.

File: peripherals/sdio/sdio_f4.yaml [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+  RESP3:
+    CARDSTATUS3: [0, 0xFFFFFFFF]
+  RESP4:
+    CARDSTATUS4: [0, 0xFFFFFFFF]
+
+  DTIMER:
+    DATATIME: [0, 0xFFFFFFFF]
+
+  DLEN:
+    DATALENGTH: [0, 0xFFFFFF]
```
Comment (adamgreig) [inline peripherals/sdio/sdio_f4.yaml:77]:
```suggestion
    DATALENGTH: [0, 0x01FFFFFF]
```

File: peripherals/sdio/sdio_f4.yaml [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    RWMOD:
+      D2: [0, "Read wait control stopping using SDIO_D2"]
+      Ck: [1, "Read wait control using SDIO_CK"]
+    RWSTOP:
+      Disabled: [0, "Read wait in progress if RWSTART is enabled"]
+      Enabled: [1, "Enable for read wait stop if RWSTART is enabled"]
+    RWSTART:
+      Disabled: [0, "Don't start read wait operation"]
+      Enabled: [1, "Read wait operation starts"]
+    DBLOCKSIZE: [0, 0xFF]
```
Comment (adamgreig) [inline peripherals/sdio/sdio_f4.yaml:92]:
```suggestion
    DBLOCKSIZE: [0, 0xF]
```

File: peripherals/sdio/sdio_f4.yaml [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+      StreamMode: [1, "Stream or SDIO multibyte data transfer"]
+    DTDIR:
+      ControllerToCard: [0, "From controller to card"]
+      CardToController: [1, "From card to controller"]
+    DTEN:
+      Disabled: [0, "Disabled"]
+      Enabled: [1, "Start transfer"]
+
+  DCOUNT:
+    DATACOUNT: [0, 0x00FFFFFF]
```
Comment (adamgreig) [inline peripherals/sdio/sdio_f4.yaml:107]:
```suggestion
    DATACOUNT: [0, 0x01FFFFFF]
```

**Comment by jkristell** [comment]
Thank you for the review, and sorry for the field length errors, I should have double checked them. Will do that next time!

I updated the branch with your fixes and ran matchperipheral.py and let it add the peripheral to the devieces and added the FIFOData field to the SDIO_FIFO.

I didn't plan to bother with the interrupt register fields. As they are all bit fields, I didn't think it was worth the effort. But I can be convinced to do a follow up patch.

**Comment by adamgreig** [comment]
Great, thanks!

> I didn't plan to bother with the interrupt register fields. As they are all bit fields, I didn't think it was worth the effort. But I can be convinced to do a follow up patch.

Since they are all simple fields you might be able to do something very simple like:

```yaml
SDIO_MASK:
    "*IE":
        Disabled: [0, "Interrupt disabled"]
        Enabled: [1, "Interrupt enabled"]
SDIO_ICR:
    "*C":
         _write:
              Clear: [1, "Clear flag"]
```

I think it might be worth doing the STA register fully though as I expect a lot of driver code will regularly be polling that register so it's much nicer if they can use the enumerated values (e.g. rxdavl().data_available() rather than rxdavl().bit_set()).

**Comment by jkristell** [comment]
Will try to find some time to look at this later this week. So lets keep this open for the time being.

**Comment by jkristell** [comment]
Updated with interrupt field value documentation.

File: peripherals/sdio/sdio_f4.yaml [hunk 4]
```diff
@@ -106,6 +106,90 @@ SDIO:
   DCOUNT:
     DATACOUNT: [0, 0x01FFFFFF]
 
+  STA:
+    CEATAEND:
+      NotReceiver: [0, "Completion signal not received"]
```
Comment (adamgreig) [inline peripherals/sdio/sdio_f4.yaml:111]:
```suggestion
      NotReceived: [0, "Completion signal not received"]
```

File: peripherals/sdio/sdio_f4.yaml [hunk 5]
```diff
@@ -106,6 +106,90 @@ SDIO:
...
+      Full: [1, "Receive FIFO full. When HW Flow Control is enabled, RXFIFOF signals becomes activated 2 words before the FIFO is full."]
+    TXFIFOF:
+      NotFull: [0, "Transmit FIFO not full"]
+      Full: [1, "Transmit FIFO full"]
+    RXFIFOHF:
+      NotHalfFull: [0, "Receive FIFO not half full"]
+      HalfFull: [1, "Receive FIFO half full. At least 8 words in the FIFO"]
+    TXFIFOHE:
+      NotHalfFull: [0, "Transmit FIFO not half full"]
+      HalfFull: [1, "Transmit FIFO half empty. At least 8 words can be written into the FIFO"]
```
Comment (adamgreig) [inline peripherals/sdio/sdio_f4.yaml:139]:
```suggestion
      NotHalfEmpty: [0, "Transmit FIFO not half empty"]
      HalfEmpty: [1, "Transmit FIFO half empty. At least 8 words can be written into the FIFO"]
```

File: peripherals/sdio/sdio_f4.yaml [hunk 6]
```diff
@@ -106,6 +106,90 @@ SDIO:
...
+      InProgress: [1, "Data receive in progress"]
+    TXACT:
+      NotInProgress: [0, "Data transmit is not in progress"]
+      InProgress: [1, "Data transmit in progress"]
+    CMDACT:
+      NotInProgress: [0, "Command transfer not in progress"]
+      InProgress: [1, "Command tranfer in progress"]
+    DBCKEND:
+      NotTransfered: [0, "Data block not sent/received (CRC check failed)"]
+      Transfered: [1, "Data block sent/received (CRC check passed)"]
```
Comment (adamgreig) [inline peripherals/sdio/sdio_f4.yaml:151]:
```suggestion
      NotTransferred: [0, "Data block not sent/received (CRC check failed)"]
      Transferred: [1, "Data block sent/received (CRC check passed)"]
```

**Pull request merged**